scio-parquet/src/main/scala/com/spotify/scio/parquet/avro/ParquetAvroIO.scala

```diff
@@ -159,38 +169,102 @@ object ParquetAvroIO {
     val readSchema: Schema =
       if (isSpecific) ReflectData.get().getSchema(avroClass) else projection
 
-    def read(sc: ScioContext, path: String): SCollection[T] = {
-      val hadoopConf = Option(conf).getOrElse(new Configuration())
+    def read(sc: ScioContext, path: String)(implicit coder: Coder[T]): SCollection[T] = {
```

Comment:
Can't we set the coder as implicit at the class level, where the type parameter is ?

Reply:
I thought about that.... This code was set up really oddly -- the read logic is implemented within the `ReadParam` case class -- and I thought from a design perspective, it's extra strange to make a ReadParam[T] require an implicit T: Coder (plus, could be a breaking change from the user perspective if they're constructing a ReadParam outside the Scio job class?). So, this felt like the least weird way to implement this, lol.